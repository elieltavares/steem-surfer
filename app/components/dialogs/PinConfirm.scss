@import '../../style/mixins';
@import '../../style/colors';

.pin-confirm-dialog-content {
  padding: 35px;

  .dialog-form {
    align-items: center;
    display: flex;
    flex-wrap: wrap;

    .brand {
      height: 128px;
      margin-right: 12px;
      width: 128px;

      img {
        width: 100%;
      }
    }

    .form-content {
      padding-top: 14px;
      width: calc(100% - 140px);
    }

    .form-content {
      .last-change {
        margin: 1em 0;
      }

      .invalidate {
        color: $silver;
        margin: 1em 0;
        display: table;
        text-decoration: underline;
      }
    }
  }
}
