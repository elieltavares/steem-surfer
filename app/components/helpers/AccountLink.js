import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getAccount } from '../../backend/steem-client';

export const makePath = username => `/@${username}`;

class AccountLink extends Component {
  goProfile = async () => {
    const { username, history, actions, onClick, afterClick } = this.props;

    onClick();

    let { accountData } = this.props;
    const { setVisitingAccount } = actions;

    if (!accountData) {
      try {
        accountData = await getAccount(username);
      } catch (err) {
        accountData = null;
      }
    }

    if (accountData) {
      setVisitingAccount(accountData);
    }

    history.push(makePath(username));

    afterClick();
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goProfile
    });
  }
}

AccountLink.defaultProps = {
  accountData: null,
  onClick: () => {},
  afterClick: () => {}
};

AccountLink.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  accountData: PropTypes.instanceOf(Object),
  onClick: PropTypes.func,
  afterClick: PropTypes.func,
  actions: PropTypes.shape({
    setVisitingAccount: PropTypes.func.isRequired
  }).isRequired
};

export default AccountLink;
