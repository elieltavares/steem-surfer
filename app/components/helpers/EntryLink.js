import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getContent } from '../../backend/steem-client';

export const makePath = (category, author, permlink, toReplies = false) =>
  `/${category}/@${author}/${permlink}${toReplies ? '#replies' : ''}`;

class EntryLink extends Component {
  goEntry = async () => {
    let { entry } = this.props;
    const { history, actions, toReplies, markAsRead } = this.props;

    if (markAsRead) {
      markAsRead();
    }

    if (!entry) {
      const { author, permlink } = this.props;
      entry = await getContent(author, permlink);
    }

    if (entry) {
      const { category, author, permlink } = entry;
      actions.setVisitingEntry(entry);
      history.push(makePath(category, author, permlink, toReplies));
    }
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goEntry
    });
  }
}

EntryLink.defaultProps = {
  entry: null,
  toReplies: false,
  markAsRead: null
};

EntryLink.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  entry: PropTypes.shape({
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    permlink: PropTypes.string.isRequired
  }),
  author: PropTypes.string.isRequired,
  permlink: PropTypes.string.isRequired,
  toReplies: PropTypes.bool,
  actions: PropTypes.shape({
    setVisitingEntry: PropTypes.func.isRequired
  }).isRequired,
  markAsRead: PropTypes.func
};

export default EntryLink;
