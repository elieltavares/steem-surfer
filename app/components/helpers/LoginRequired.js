import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'antd';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';

import Login from '../dialogs/Login';

export const checkLogin = (activeAccount, requiredKeys) => {
  // Steem connect login.
  if (activeAccount && activeAccount.type === 'sc') {
    return true;
  }

  // Traditional login. Check all required keys exists in user keys.
  if (
    activeAccount &&
    activeAccount.type === 's' &&
    requiredKeys.every(e => activeAccount.keys[e])
  ) {
    return true;
  }

  return false;
};

class LoginRequired extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loginModalVisible: false
    };
  }

  showLoginModal = () => {
    this.setState({
      loginModalVisible: true
    });

    setTimeout(() => {
      const { onDialogOpen } = this.props;
      onDialogOpen();
    }, 200);
  };

  onLoginModalCancel = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  onLoginSuccess = () => {
    this.setState({
      loginModalVisible: false
    });

    setTimeout(() => {
      if (!this.check()) {
        this.showLoginModal();
      }
    }, 400);
  };

  afterModalClose = () => {
    const { onDialogClose } = this.props;
    onDialogClose();
  };

  check = () => {
    const { activeAccount, requiredKeys } = this.props;

    return checkLogin(activeAccount, requiredKeys);
  };

  render() {
    const { children } = this.props;

    if (this.check()) {
      return children;
    }

    const { activeAccount, requiredKeys } = this.props;
    const { loginModalVisible } = this.state;

    // Default message
    let loginMsg = <FormattedMessage id="login-required.default" />;

    if (activeAccount) {
      // More specific, key based message
      loginMsg =
        requiredKeys.length > 1 ? (
          <FormattedHTMLMessage
            id="login-required.keys-required"
            values={{ keys: requiredKeys }}
          />
        ) : (
          <FormattedHTMLMessage
            id="login-required.key-required"
            values={{ key: requiredKeys }}
          />
        );
    }

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.showLoginModal();
      }
    });

    return (
      <Fragment>
        {newChild}

        <Modal
          visible={loginModalVisible}
          onCancel={this.onLoginModalCancel}
          footer={false}
          width="500px"
          closable={false}
          destroyOnClose
          centered
          afterClose={this.afterModalClose}
        >
          <Login
            {...this.props}
            onSuccess={this.onLoginSuccess}
            loginMsg={loginMsg}
          />
        </Modal>
      </Fragment>
    );
  }
}

LoginRequired.defaultProps = {
  activeAccount: null,
  onDialogOpen: () => {},
  onDialogClose: () => {},
  requiredKeys: []
};

LoginRequired.propTypes = {
  children: PropTypes.element.isRequired,
  activeAccount: PropTypes.instanceOf(Object),
  requiredKeys: PropTypes.arrayOf(PropTypes.string),
  onDialogOpen: PropTypes.func,
  onDialogClose: PropTypes.func
};

export default LoginRequired;
