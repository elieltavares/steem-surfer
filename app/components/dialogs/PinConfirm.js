import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button, Col, Input, Popconfirm, Alert, message } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';

import { pinHasher } from '../../utils/crypto';
import logo from '../../img/logo-big.png';
import defaults from '../../constants/defaults';
import { getItem, removeItem } from '../../helpers/storage';

class PinConfirm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: '',
      trys: 1
    };
  }

  componentDidMount() {
    document.addEventListener('click', this.docClicked);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.docClicked);
  }

  docClicked = e => {
    if (e.target.id !== 'pin-input') {
      document.querySelector('#pin-input').focus();
    }
  };

  handleChange = event => {
    this.setState({ value: event.target.value });
  };

  invalidate = () => {
    const { intl } = this.props;
    const { actions, onInvalidate, history } = this.props;

    history.push('/');

    setTimeout(() => {
      actions.wipePin();
      removeItem('pin-code');
      actions.logOut();
      actions.deleteAccounts();
      onInvalidate();
      message.info(intl.formatMessage({ id: 'confirm-pin-code.invalidated' }));
    }, 100);
  };

  submitForm = e => {
    if (e) e.stopPropagation();

    const { onSuccess } = this.props;
    const { value, trys } = this.state;

    if (trys === defaults.maxPinCodeTry) {
      this.invalidate();
      return;
    }

    const compareHash = getItem('pin-code');
    const hashed = pinHasher(value);

    if (hashed === compareHash) {
      onSuccess(value);
      return;
    }

    this.setState({ value: '', trys: trys + 1 });
  };

  render() {
    const { intl } = this.props;
    const { value, trys } = this.state;

    return (
      <div className="pin-confirm-dialog-content">
        <div className="dialog-form">
          <div className="brand">
            <img src={logo} alt="" />
          </div>
          <div className="form-content">
            <h2 className="form-title">
              <FormattedMessage id="confirm-pin-code.title" />
            </h2>
            <Input.Group>
              <Col span={18}>
                <Input
                  autoFocus
                  type="password"
                  id="pin-input"
                  maxLength={20}
                  placeholder={intl.formatMessage({
                    id: 'confirm-pin-code.input-placeholder'
                  })}
                  value={value}
                  onChange={this.handleChange}
                  onKeyUp={e => {
                    if (e.key === 'Enter') {
                      this.submitForm();
                    }
                  }}
                />
              </Col>
              <Col span={3}>
                <Button
                  type="primary"
                  shape="circle"
                  disabled={value.length < 1}
                  onClick={this.submitForm}
                >
                  <i className="mi">lock_open</i>
                </Button>
              </Col>
            </Input.Group>

            {trys === defaults.maxPinCodeTry && (
              <div className="last-change">
                <Alert
                  message={intl.formatMessage({
                    id: 'confirm-pin-code.last-chance'
                  })}
                  type="warning"
                  showIcon
                />
              </div>
            )}

            {trys > 1 && (
              <Popconfirm
                title={intl.formatMessage({
                  id: 'confirm-pin-code.invalidate-confirm'
                })}
                onConfirm={this.invalidate}
                okText={intl.formatMessage({ id: 'g.yes' })}
                cancelText={intl.formatMessage({ id: 'g.no' })}
              >
                <a className="invalidate">
                  <FormattedMessage id="confirm-pin-code.invalidate" />
                </a>
              </Popconfirm>
            )}
          </div>
        </div>
      </div>
    );
  }
}

PinConfirm.defaultProps = {
  onInvalidate: () => {}
};

PinConfirm.propTypes = {
  onSuccess: PropTypes.func.isRequired,
  onInvalidate: PropTypes.func,
  actions: PropTypes.shape({
    wipePin: PropTypes.func.isRequired,
    logOut: PropTypes.func.isRequired,
    deleteAccounts: PropTypes.func.isRequired
  }).isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  intl: PropTypes.instanceOf(Object).isRequired
};

export default injectIntl(PinConfirm);
