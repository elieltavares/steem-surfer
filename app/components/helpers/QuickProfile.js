/*
eslint-disable import/no-cycle
*/

import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Drawer, message } from 'antd';
import { FormattedMessage, FormattedNumber } from 'react-intl';

import { proxifyImageSrc } from '@esteemapp/esteem-render-helpers';

import Tooltip from '../common/Tooltip';

import {
  getAccount,
  getDiscussions,
  getFollowCount
} from '../../backend/steem-client';
import authorReputation from '../../utils/author-reputation';

import UserAvatar from '../elements/UserAvatar';
import EntryListLoadingItem from '../elements/EntryListLoadingItem';
import LinearProgress from '../common/LinearProgress';
import EntryListItem from '../elements/EntryListItem';
import FollowControls from '../elements/FollowControls';
import AccountLink from './AccountLink';
import coverFallbackDay from '../../img/cover-fallback-day.png';
import coverFallbackNight from '../../img/cover-fallback-night.png';

class QuickProfile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
      active: false,
      profile: {
        name: ' ',
        about: ' ',
        coverImage: null,
        postCount: 0
      },
      follows: {
        followerCount: null,
        followingCount: null
      },
      loadingEntries: true,
      entries: [],
      accountData: null
    };
  }

  load = async () => {
    const { username, intl } = this.props;

    const account = await getAccount(username);

    this.setState({ accountData: account });

    let accountProfile;
    try {
      accountProfile = JSON.parse(account.json_metadata).profile;
    } catch (err) {
      accountProfile = {};
    }

    if (accountProfile) {
      const name = accountProfile.name || null;
      const about = accountProfile.about || null;
      const postCount = account.post_count;
      const coverImage = accountProfile.cover_image || null;

      this.setState({ profile: { name, about, coverImage, postCount } });
    }

    let follow;
    try {
      follow = await getFollowCount(username);
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-profile' })
      );
    }

    if (follow) {
      const followerCount = follow.follower_count;
      const followingCount = follow.following_count;

      this.setState({ follows: { followerCount, followingCount } });
    }

    let entries;
    try {
      entries = await getDiscussions('blog', {
        tag: username,
        limit: 7,
        start_author: undefined,
        start_permlink: undefined
      });
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-content' })
      );
    }

    if (entries) {
      this.setState({ entries, loadingEntries: false });
    }
  };

  show = () => {
    this.setState({ active: true, visible: true });
    this.load();
  };

  hide = () => {
    this.setState({ visible: false });
    setTimeout(() => {
      this.setState({ active: false });
    }, 500);
  };

  goSection = section => {
    const { history, username } = this.props;
    const u = `/@${username}/${section}`;
    history.push(u);
  };

  render() {
    const { children, username, global, intl } = this.props;
    let { reputation } = this.props;
    const {
      visible,
      active,
      profile,
      follows,
      entries,
      loadingEntries,
      accountData
    } = this.state;

    if (accountData) ({ reputation } = accountData);

    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.show();
      }
    });

    const followersMsg =
      follows.followerCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followerCount} />
      );
    const postCountMsg =
      profile.postCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={profile.postCount} />
      );
    const followingMsg =
      follows.followingCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followingCount} />
      );

    const bgImage =
      (profile.coverImage && proxifyImageSrc(profile.coverImage)) ||
      (global.theme === 'day' ? coverFallbackDay : coverFallbackNight);

    return (
      <Fragment>
        {newChild}

        {active && (
          <Drawer
            placement="right"
            closable={false}
            onClose={this.hide}
            visible={visible}
            width="640px"
          >
            <div
              className={`quick-profile-content ${
                loadingEntries ? 'loading' : ''
              } `}
            >
              <div className="profile-area">
                <div
                  className="account-cover"
                  style={{ backgroundImage: `url('${bgImage}')` }}
                >
                  <div className="profile-avatar">
                    <UserAvatar {...this.props} user={username} size="large" />
                    <div className="reputation">
                      {authorReputation(reputation)}
                    </div>
                  </div>
                  <div className="follow-btn-holder">
                    <FollowControls {...this.props} targetUsername={username} />
                  </div>
                </div>
                <AccountLink
                  {...this.props}
                  username={username}
                  accountData={accountData}
                >
                  <div className="username">{username}</div>
                </AccountLink>
                <div className="about">{profile.about}</div>
                <div className="numbers">
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.post-count'
                    })}
                  >
                    <span className="post-count">
                      <i className="mi">list</i>
                      {postCountMsg}
                    </span>
                  </Tooltip>
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.followers'
                    })}
                  >
                    <span className="followers">
                      <i className="mi">people</i>
                      {followersMsg}
                    </span>
                  </Tooltip>
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.following'
                    })}
                  >
                    <span className="following">
                      <i className="mi">person_add</i>
                      {followingMsg}
                    </span>
                  </Tooltip>
                </div>
              </div>

              <div className="account-menu">
                <div className="account-menu-items">
                  <a
                    role="none"
                    className="menu-item selected-item"
                    onClick={() => {
                      this.goSection('blog');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-blog" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('comments');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-comments" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('replies');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-replies" />
                  </a>
                  <span className="separator-item" />
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('points');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-points" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('wallet');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-wallet" />
                  </a>
                </div>
              </div>

              {loadingEntries && (
                <Fragment>
                  <LinearProgress />
                  <div className="entries">
                    <EntryListLoadingItem />
                  </div>
                </Fragment>
              )}

              {!loadingEntries && (
                <div className="entries">
                  {entries.map(d => (
                    <EntryListItem
                      key={`${d.author}-${d.permlink}`}
                      {...this.props}
                      entry={d}
                      asAuthor={username}
                      inDrawer
                    />
                  ))}
                </div>
              )}
            </div>
          </Drawer>
        )}
      </Fragment>
    );
  }
}

QuickProfile.defaultProps = {};

QuickProfile.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  reputation: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  global: PropTypes.shape({
    theme: PropTypes.string.isRequired
  }).isRequired,
  intl: PropTypes.instanceOf(Object).isRequired
};

export default QuickProfile;
