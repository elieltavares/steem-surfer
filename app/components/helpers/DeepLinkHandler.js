import React from 'react';
import PropTypes from 'prop-types';

import filters from '../../constants/filters.json';

import { makePath as makePathTag } from './TagLink';
import { makePath as makePathAccount } from './AccountLink';
import { makePath as makePathEntry } from './EntryLink';

export const deepLink2Obj = url => {
  let urlPart = url.split('://')[1];

  // remove last char if /
  if (urlPart.endsWith('/')) {
    urlPart = urlPart.substring(0, urlPart.length - 1);
  }

  const parts = urlPart.split('/');

  // witnesses
  if (parts.length === 1 && parts[0] === 'witnesses') {
    return { type: 'witnesses' };
  }

  // sps
  if (parts.length === 1 && parts[0] === 'sps') {
    return { type: 'sps' };
  }

  // filter
  if (parts.length === 1 && filters.includes(parts[0])) {
    return { type: 'filter', filter: parts[0] };
  }

  // filter with tag
  if (parts.length === 2 && filters.includes(parts[0])) {
    return { type: 'filter-tag', filter: parts[0], tag: parts[1] };
  }

  // account
  if (parts.length === 1 && parts[0].startsWith('@')) {
    return { type: 'account', account: parts[0].replace('@', '') };
  }

  // post without tag
  if (parts.length === 2 && parts[0].startsWith('@')) {
    return {
      type: 'post',
      cat: 'esteem',
      author: parts[0].replace('@', ''),
      permlink: parts[1]
    };
  }

  // post with tag
  if (parts.length === 3 && parts[1].startsWith('@')) {
    return {
      type: 'post',
      cat: parts[0],
      author: parts[1].replace('@', ''),
      permlink: parts[2]
    };
  }
};

class DeepLinkHandler extends React.Component {
  componentDidMount() {
    window.addEventListener('deep-link', this.handler);
  }

  componentWillUnmount() {
    window.removeEventListener('deep-link', this.handler);
  }

  handler = e => {
    const { history } = this.props;
    const { url } = e.detail;
    const obj = deepLink2Obj(url);

    let path;

    switch (obj.type) {
      case 'filter':
        path = `/${obj.filter}`;
        break;
      case 'filter-tag':
        path = makePathTag(obj.filter, obj.tag);
        break;
      case 'account':
        path = makePathAccount(obj.account);
        break;
      case 'post':
        path = makePathEntry(obj.cat, obj.author, obj.permlink);
        break;
      case 'sps':
        path = `/sps`;
        break;
      case 'witnesses':
        path = `/witnesses`;
        break;
      default:
        path = null;
        break;
    }

    if (path) {
      history.push(path);
    }
  };

  render() {
    return null;
  }
}

DeepLinkHandler.propTypes = {
  history: PropTypes.shape({}).isRequired
};

export default DeepLinkHandler;
