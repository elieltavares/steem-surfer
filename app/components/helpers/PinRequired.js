import React, { Component, Fragment } from 'react';
import { Modal } from 'antd';

import PropTypes from 'prop-types';
import PinConfirm from '../dialogs/PinConfirm';

class PinRequired extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false
    };
  }

  onPinInvalidated = () => {
    this.setState({ show: false });
  };

  onConfirmPinSuccess = value => {
    const { onSuccess } = this.props;
    this.setState({ show: false }, () => {
      onSuccess(value);
    });
  };

  cancelFn = () => {
    this.setState({ show: false });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.setState({ show: true });
      }
    });

    return (
      <Fragment>
        {show && (
          <Modal
            footer={null}
            closable
            onCancel={this.cancelFn}
            keyboard
            visible
            width="500px"
            centered
            destroyOnClose
          >
            <PinConfirm
              {...this.props}
              onSuccess={this.onConfirmPinSuccess}
              onInvalidate={this.onPinInvalidated}
            />
          </Modal>
        )}
        {newChild}
      </Fragment>
    );
  }
}

PinRequired.defaultProps = {
  onSuccess: () => {}
};

PinRequired.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  onSuccess: PropTypes.func
};

export default PinRequired;
