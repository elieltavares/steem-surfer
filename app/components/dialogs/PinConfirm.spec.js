/* eslint-disable */

import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PinConfirm from './PinConfirm';
import { mountWithIntl } from '../../helpers/intl-test';
import PropTypes from 'prop-types';

Enzyme.configure({ adapter: new Adapter() });

describe('PinConfirm', () => {
  it('(1) Render', () => {
    const props = {
      onSuccess: () => {},
      invalidateFn: () => {},
      actions: {
        wipePin: () => {},
        logOut: () => {},
        deleteAccounts: () => {}
      },
      compareHash: 'foo',
      history: {}
    };

    const wrapper = mountWithIntl(<PinConfirm {...props} />);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
