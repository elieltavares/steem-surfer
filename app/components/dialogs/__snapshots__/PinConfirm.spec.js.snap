// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinConfirm (1) Render 1`] = `"<div class=\\"pin-confirm-dialog-content\\"><div class=\\"dialog-form\\"><div class=\\"brand\\"><img src=\\"test-file-stub\\" alt=\\"\\"></div><div class=\\"form-content\\"><h2 class=\\"form-title\\"><span>Enter PIN Code</span></h2><span class=\\"ant-input-group\\"><div class=\\"ant-col-18\\"><input type=\\"password\\" id=\\"pin-input\\" maxlength=\\"20\\" placeholder=\\"Enter PIN code to continue\\" class=\\"ant-input\\" value=\\"\\"></div><div class=\\"ant-col-3\\"><button disabled=\\"\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary ant-btn-circle\\"><i class=\\"mi\\">lock_open</i></button></div></span></div></div></div>"`;
